### PR TITLE
lparstat.py: Add lparstat tests

### PR DESCRIPTION
**Pull request by nasastry**
Test lparstat command with a list of correct command line arguments and
in-correct command line arguments

Signed-off-by: Nageswara R Sastry <rnsastry@linux.ibm.com>

### CODE REVIEW

**Comment by nasastry** [comment]
```
# avocado run lparstat.py -m lparstat.py.data/options.yaml
JOB ID     : f9fadeed5007b686c954d3da818d6145b057bbac
JOB LOG    : /home/avocado-fvt-wrapper/results/job-2021-07-26T03.14-f9fadee/job.log
 (1/2) lparstat.py:lparstat.test_list;run-df45: PASS (11.17 s)
 (2/2) lparstat.py:lparstat.test_nlist;run-df45: PASS (1.02 s)
RESULTS    : PASS 2 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/avocado-fvt-wrapper/results/job-2021-07-26T03.14-f9fadee/results.html
JOB TIME   : 12.77 s

# avocado run lparstat.py
JOB ID     : 21cdbd9cd0ce5bace9eb0123691caee8bb6e93e7
JOB LOG    : /home/avocado-fvt-wrapper/results/job-2021-07-26T03.15-21cdbd9/job.log
 (1/2) lparstat.py:lparstat.test_list: PASS (3.11 s)
 (2/2) lparstat.py:lparstat.test_nlist: PASS (1.07 s)
RESULTS    : PASS 2 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/avocado-fvt-wrapper/results/job-2021-07-26T03.15-21cdbd9/results.html
JOB TIME   : 4.72 s
```

**Review by PraveenPenguin — APPROVED** [review]
LGTM

File: ras/lparstat.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+                self.fail("lparstat: %s command failed to execute" % cmd)
+
+    def test_nlist(self):
+        # Negative tests
+        lists = self.params.get('nlist', default=['--nonexistingoption'])
+        for list_item in lists:
+            cmd = "lparstat %s" % list_item
+            if not process.system(cmd, ignore_status=True, sudo=True):
+                self.log.info("%s command failed" % cmd)
+                self.fail("lparstat: %s command failed to execute" % cmd)
```
Comment (sacsant) [inline ras/lparstat.py:45]: This fail message does not seem correct. A better message might be something like
"Expected failure, but command executed successfully"

Comment (nasastry) [reply ras/lparstat.py]: @sacsant nice, catch. Thanks!! modified the comment. please review.

**Comment by nasastry** [comment]
```
# avocado run lparstat.py
JOB ID     : 22735c82d5331064eac43536913ffe6e209cfdc2
JOB LOG    : /home/avocado-fvt-wrapper/results/job-2021-07-27T08.05-22735c8/job.log
 (1/2) lparstat.py:lparstat.test_list: PASS (3.15 s)
 (2/2) lparstat.py:lparstat.test_nlist: PASS (1.09 s)
RESULTS    : PASS 2 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/avocado-fvt-wrapper/results/job-2021-07-27T08.05-22735c8/results.html
JOB TIME   : 4.76 s

# avocado run lparstat.py -m lparstat.py.data/options.yaml
JOB ID     : eb19a774cb77038edb1702dc572e598d4e3b902c
JOB LOG    : /home/avocado-fvt-wrapper/results/job-2021-07-27T08.06-eb19a77/job.log
 (1/2) lparstat.py:lparstat.test_list;run-df45: PASS (11.13 s)
 (2/2) lparstat.py:lparstat.test_nlist;run-df45: PASS (1.11 s)
RESULTS    : PASS 2 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/avocado-fvt-wrapper/results/job-2021-07-27T08.06-eb19a77/results.html
JOB TIME   : 12.87 s
```

File: ras/lparstat.py (hunk 1 above)
Comment (sacsant) [reply ras/lparstat.py]: looks good.

**Comment by sacsant** [comment]
@nasastry latest version of powerpc-utils add a new option -x to lparstat (to display security settings). You may want to either fold that into this patch or submit a new patch to test -x option

https://github.com/ibm-power-utilities/powerpc-utils/commit/97269d301797e23b75d0c7a5cb63ce280783f615

**Comment by nasastry** [comment]
> @nasastry latest version of powerpc-utils add a new option -x to lparstat (to display security settings). You may want to either fold that into this patch or submit a new patch to test -x option
> 
> [ibm-power-utilities/powerpc-utils@97269d3](https://github.com/ibm-power-utilities/powerpc-utils/commit/97269d301797e23b75d0c7a5cb63ce280783f615)

@sacsant Already added -x option with the original PR
```
in yaml file:
list: ['-h', '-V', '--version', '-i', '-x', '-E', '-l', '--legacy', "1 10"]

in .py file:
lists = self.params.get('list', default=['-i', '-x', '-E', '-l', '1 2'])
```

Thanks for notifying.

**Pull request merged**